flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/AppendOnlyWriter.java

```diff
@@ -0,0 +1,205 @@
+    private final Comparator<RowData> keyComparator;
+    private final RowDataSerializer keySerializer;
+    private final List<DataFileMeta> existingFiles;
+    private final List<DataFileMeta> newFiles;
+    private final Supplier<Path> pathCreator;
+    private final FileStatsExtractor fileStatsExtractor;
+    private final FieldStatsCollector fieldStatsCollector;
+
+    private final AtomicLong startingSeqNum;
+    private final AtomicLong nextSeqNum;
+    private final AtomicInteger rowCount;
+
+    private final AtomicReference<BinaryRowData> minKeyRef = new AtomicReference<>(null);
+    private final AtomicReference<BinaryRowData> maxKeyRef = new AtomicReference<>(null);
+
+    public AppendOnlyWriter(
+            FileFormat fileFormat,
+            long targetFileSize,
+            RowType writeSchema,
+            Comparator<RowData> keyComparator,
+            List<DataFileMeta> existingFiles,
+            Supplier<Path> pathCreator,
+            FileStatsExtractor fileStatsExtractor) {
+
+        this.writerFactory = fileFormat.createWriterFactory(writeSchema);
+
+        this.writer = new BaseRollingFileWriter(writerFactory, targetFileSize, pathCreator);
+        this.writer.register(new CloseWriterListener());
+
+        this.targetFileSize = targetFileSize;
+
+        this.keyComparator = keyComparator;
+        this.keySerializer = new RowDataSerializer(writeSchema);
+
+        this.existingFiles = existingFiles;
+        this.newFiles = new ArrayList<>();
+        this.pathCreator = pathCreator;
+        this.fileStatsExtractor = fileStatsExtractor;
+        this.fieldStatsCollector =
+                fileStatsExtractor == null ? new FieldStatsCollector(writeSchema) : null;
+
+        this.startingSeqNum = new AtomicLong(0L);
+        this.nextSeqNum = new AtomicLong(maxSequenceNumber() + 1);
+        this.rowCount = new AtomicInteger(0);
+    }
+
+    private long maxSequenceNumber() {
+        return existingFiles.stream()
+                .map(DataFileMeta::maxSequenceNumber)
+                .max(Long::compare)
+                .orElse(-1L);
+    }
+
+    @Override
+    public void append(RowData row) throws Exception {
+        writer.write(row);
+        if (fieldStatsCollector != null) {
+            fieldStatsCollector.collect(row);
+        }
+        updateMinKey(row);
+        updateMaxKey(row);
+
+        rowCount.incrementAndGet();
+        nextSeqNum.incrementAndGet();
+    }
+
+    private void updateMinKey(RowData row) {
+        BinaryRowData minKey = minKeyRef.get();
+        if ((minKey == null) || (keyComparator.compare(minKey, row) > 0)) {
+            minKey = keySerializer.toBinaryRow(row);
+        }
+        minKeyRef.set(minKey);
+    }
+
+    private void updateMaxKey(RowData row) {
+        BinaryRowData maxKey = maxKeyRef.get();
+        if ((maxKey == null) || (keyComparator.compare(maxKey, row) < 0)) {
+            maxKey = keySerializer.toBinaryRow(row);
+        }
+        maxKeyRef.set(maxKey);
```

Comment:
That's quite time-consuming for the append-only table, because we will need to compare the whole row each time to get the max key and min key. 

It's better to set the dummy constant key and set the whole row as the value, then we can skip to compare the whole row columns to get the max-min keys ( Because the key is always a constant).

Reply:
We can also skip the key value serialization for each encountered row...